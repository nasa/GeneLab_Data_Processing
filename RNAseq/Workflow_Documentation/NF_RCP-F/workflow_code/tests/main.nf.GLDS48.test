nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "core"

    test("GLDS-48:Mouse,SingleEnd,NonERCC: Should run without failures") {

        when {
            params {
                gldsAccession = 'GLDS-48' // GeneLab Data Accession Number, e.g. GLDS-104
                use_dummy_gene_counts = true // Use random gene counts for Deseq2, this addresses an issue where low/zero gene counts causes DGE analysis to fail
                genomeSubsample = 19 // Subsamples the reference fasta and gtf to a single sequence (often representing a single chromosome)
                truncateTo = 300 // Subsamples the raw reads files to the specified number of reads for EACH raw reads file.
            }
        }

        then {
            def TEST_DATASET = "GLDS-48"
            def TEST_SAMPLE_1 = "Mmus_C57-6J_LVR_FLT_C_Rep1_M25"
            def TEST_SAMPLE_2 = "Mmus_C57-6J_LVR_GC_C_Rep1_M36"
            def TEST_SAMPLE_3 = "Mmus_C57-6J_LVR_GC_C_Rep5_M40"
            def TEST_SAMPLE_4 = "Mmus_C57-6J_LVR_GC_I_Rep1_M31"
            def TEST_SAMPLE_5 = "Mmus_C57-6J_LVR_GC_I_Rep2_M32"
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${ TEST_DATASET }/Metadata").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/00-RawData/Fastq").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/01-TG_Preproc/Fastq").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/02-STAR_Alignment/STAR_NumNonZeroGenes.csv"),
                            path("${params.outputDir}/${ TEST_DATASET }/02-STAR_Alignment/STAR_Unnormalized_Counts.csv"),
                            path("${params.outputDir}/${ TEST_DATASET }/03-RSEM_Counts/${ TEST_SAMPLE_2 }.genes.results"),
                            path("${params.outputDir}/${ TEST_DATASET }/03-RSEM_Counts/${ TEST_SAMPLE_2 }.isoforms.results"),
                            path("${params.outputDir}/${ TEST_DATASET }/03-RSEM_Counts/RSEM_NumNonZeroGenes.csv"),
                            path("${params.outputDir}/${ TEST_DATASET }/03-RSEM_Counts/RSEM_Unnormalized_Counts.csv"),
                            path("${params.outputDir}/${ TEST_DATASET }/04-DESeq2_NormCounts").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/05-DESeq2_DGE").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/GeneLab").list(),
                            path("${params.outputDir}/${ TEST_DATASET }/RSeQC_Analyses/02_geneBody_coverage/${ TEST_SAMPLE_3 }/${ TEST_SAMPLE_3 }.geneBodyCoverage.txt"),
                            path("${params.outputDir}/${ TEST_DATASET }/RSeQC_Analyses/03_infer_experiment/${ TEST_SAMPLE_3 }_infer_expt.out"),
                            // path("${params.outputDir}/${ TEST_DATASET }/RSeQC_Analyses/04_inner_distance/${ TEST_SAMPLE_4 }/${ TEST_SAMPLE_4 }.inner_distance.txt"),
                            path("${params.outputDir}/${ TEST_DATASET }/RSeQC_Analyses/05_read_distribution/${ TEST_SAMPLE_5 }_read_dist.out"),
                            ).match()
        }
    }
}
