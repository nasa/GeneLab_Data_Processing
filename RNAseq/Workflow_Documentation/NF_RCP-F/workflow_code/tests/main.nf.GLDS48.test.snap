{
    "GLDS-48: Should run without failures": {
        "content": [
            [
                "GLDS-48_bulkRNASeq_v1_runsheet.csv:md5,cf541f39caa08636622d9d17231a3969",
                "GLDS-48_metadata_table.txt:md5,4191933268365f9138d176c6e1c85147",
                "OSD-48_metadata_OSD-48-ISA.zip:md5,9f1701b3bc6bac0075cf121c2214b45b"
            ],
            [
                "Mmus_C57-6J_LVR_FLT_C_Rep1_M25_raw.fastq.gz:md5,3dced236b39fc99c33b2a47a2b867b55",
                "Mmus_C57-6J_LVR_FLT_C_Rep2_M26_raw.fastq.gz:md5,6ea9d7b67a34d5819a2b79c288ff5f69",
                "Mmus_C57-6J_LVR_FLT_C_Rep3_M27_raw.fastq.gz:md5,02ca1363e7bd005fc3e1a5355c1dfeca",
                "Mmus_C57-6J_LVR_FLT_C_Rep4_M28_raw.fastq.gz:md5,b017ac84312d1f38b7ccbb5102545c5d",
                "Mmus_C57-6J_LVR_FLT_C_Rep5_M30_raw.fastq.gz:md5,efd26062f58af6729307d2137d8fea42",
                "Mmus_C57-6J_LVR_FLT_I_Rep1_M21_raw.fastq.gz:md5,1196360a78e6c303e19b7e6bba85d8dc",
                "Mmus_C57-6J_LVR_FLT_I_Rep2_M22_raw.fastq.gz:md5,8e49e556dddc62bb06c36e20c141d291",
                "Mmus_C57-6J_LVR_GC_C_Rep1_M36_raw.fastq.gz:md5,e74158d1539869542266da6421c5da35",
                "Mmus_C57-6J_LVR_GC_C_Rep2_M37_raw.fastq.gz:md5,f51404578ae7128f53977bc845a67bd2",
                "Mmus_C57-6J_LVR_GC_C_Rep3_M38_raw.fastq.gz:md5,02a21f7484d1f640262ce6de9a590bdc",
                "Mmus_C57-6J_LVR_GC_C_Rep4_M39_raw.fastq.gz:md5,bfc55337a37f317cbaec8377a9286df9",
                "Mmus_C57-6J_LVR_GC_C_Rep5_M40_raw.fastq.gz:md5,cf57f68aa7725941ca7831ba6dafd451",
                "Mmus_C57-6J_LVR_GC_I_Rep1_M31_raw.fastq.gz:md5,5a99df7404567ba867d52728ed81c2d5",
                "Mmus_C57-6J_LVR_GC_I_Rep2_M32_raw.fastq.gz:md5,f0a0442543557b40d996de43f93b7146"
            ],
            [
                "Mmus_C57-6J_LVR_FLT_C_Rep1_M25_trimmed.fastq.gz:md5,d95ca667755c47327c2461201584d80f",
                "Mmus_C57-6J_LVR_FLT_C_Rep2_M26_trimmed.fastq.gz:md5,54089be1a8e415c9d6e3601dd895f407",
                "Mmus_C57-6J_LVR_FLT_C_Rep3_M27_trimmed.fastq.gz:md5,05d19501daf29aed84531b63b82f82a7",
                "Mmus_C57-6J_LVR_FLT_C_Rep4_M28_trimmed.fastq.gz:md5,b32116bf5e6385d26d068647e53c921f",
                "Mmus_C57-6J_LVR_FLT_C_Rep5_M30_trimmed.fastq.gz:md5,53dd7a9e5c98e001c34d7e0f0687ed32",
                "Mmus_C57-6J_LVR_FLT_I_Rep1_M21_trimmed.fastq.gz:md5,34ce059e22d5d4f0c25422f3616c7043",
                "Mmus_C57-6J_LVR_FLT_I_Rep2_M22_trimmed.fastq.gz:md5,da666ada4aa14919ea5fc0f28f5f4d12",
                "Mmus_C57-6J_LVR_GC_C_Rep1_M36_trimmed.fastq.gz:md5,1d7bfee51f4f5a487ad8359b37dbfc68",
                "Mmus_C57-6J_LVR_GC_C_Rep2_M37_trimmed.fastq.gz:md5,314d3a8977e6cb6095dc7c73c4fa0edd",
                "Mmus_C57-6J_LVR_GC_C_Rep3_M38_trimmed.fastq.gz:md5,2a854e1600889c9ef538251a7ff3988f",
                "Mmus_C57-6J_LVR_GC_C_Rep4_M39_trimmed.fastq.gz:md5,25d7415501016bdc4b7645008685affc",
                "Mmus_C57-6J_LVR_GC_C_Rep5_M40_trimmed.fastq.gz:md5,527e01a7290cb2b4799ff1495013b196",
                "Mmus_C57-6J_LVR_GC_I_Rep1_M31_trimmed.fastq.gz:md5,843427c45e8da0ea4358b2a6e334f635",
                "Mmus_C57-6J_LVR_GC_I_Rep2_M32_trimmed.fastq.gz:md5,fbf25f70a02ba4ec574b4a65300fefad"
            ],
            "Mmus_C57-6J_LVR_FLT_C_Rep1_M25_Aligned.sortedByCoord.out.bam:md5,02f04b578be6f0ee133ae4dab6961a3f",
            "Mmus_C57-6J_LVR_GC_C_Rep1_M36.genes.results:md5,0026606fc4edd86c12ce9f10dfcc94ef",
            "Mmus_C57-6J_LVR_GC_C_Rep1_M36.isoforms.results:md5,e0abe12f204c86024290f22b64c4d5f0",
            "RSEM_NumNonZeroGenes.csv:md5,d5a0492b5126e3da33aef5bf91d949b8",
            "RSEM_Unnormalized_Counts.csv:md5,1a877f2654556aa697b235e0d239ed08",
            [
                "Normalized_Counts.csv:md5,5d058c386aa82785f2e67c617b4aa409",
                "RSEM_Unnormalized_Counts.csv:md5,b30807d931a291daa6fdf70c8aee9694"
            ],
            [
                "SampleTable.csv:md5,db67009ac75a7fef281ced557352a2de",
                "contrasts.csv:md5,04924e148dc55056556ddcc89af862e7",
                "differential_expression.csv:md5,af9dcf3117e81c98a3707ebd798d4829",
                "visualization_PCA_table.csv:md5,e6af717c612842523e384e2a65d0a058",
                "visualization_output_table.csv:md5,fbeb584d9ee7592d07f4a069303c208e"
            ],
            [
                "software_versions.md:md5,d329935c7716908a5422c0e134a71531"
            ],
            "Mmus_C57-6J_LVR_GC_C_Rep5_M40.geneBodyCoverage.txt:md5,536cb073e5342fc4b2e7329e6bb130fe",
            "Mmus_C57-6J_LVR_GC_C_Rep5_M40_infer_expt.out:md5,cf4b9b80507493cd8232113620b4a765",
            "Mmus_C57-6J_LVR_GC_I_Rep2_M32_read_dist.out:md5,fe7bfe2aa0a2774fa2754ee0485c8b6e"
        ],
        "timestamp": "2023-05-09T20:33:40+0000"
    },
    "GLDS-48:Mouse,SingleEnd,NonERCC: Should run without failures": {
        "content": [
            [
                "GLDS-48_bulkRNASeq_v1_runsheet.csv:md5,cf541f39caa08636622d9d17231a3969",
                "GLDS-48_metadata_table.txt:md5,4191933268365f9138d176c6e1c85147",
                "OSD-48_metadata_OSD-48-ISA.zip:md5,9f1701b3bc6bac0075cf121c2214b45b"
            ],
            [
                "Mmus_C57-6J_LVR_FLT_C_Rep1_M25_raw.fastq.gz:md5,3dced236b39fc99c33b2a47a2b867b55",
                "Mmus_C57-6J_LVR_FLT_C_Rep2_M26_raw.fastq.gz:md5,6ea9d7b67a34d5819a2b79c288ff5f69",
                "Mmus_C57-6J_LVR_FLT_C_Rep3_M27_raw.fastq.gz:md5,02ca1363e7bd005fc3e1a5355c1dfeca",
                "Mmus_C57-6J_LVR_FLT_C_Rep4_M28_raw.fastq.gz:md5,b017ac84312d1f38b7ccbb5102545c5d",
                "Mmus_C57-6J_LVR_FLT_C_Rep5_M30_raw.fastq.gz:md5,efd26062f58af6729307d2137d8fea42",
                "Mmus_C57-6J_LVR_FLT_I_Rep1_M21_raw.fastq.gz:md5,1196360a78e6c303e19b7e6bba85d8dc",
                "Mmus_C57-6J_LVR_FLT_I_Rep2_M22_raw.fastq.gz:md5,8e49e556dddc62bb06c36e20c141d291",
                "Mmus_C57-6J_LVR_GC_C_Rep1_M36_raw.fastq.gz:md5,e74158d1539869542266da6421c5da35",
                "Mmus_C57-6J_LVR_GC_C_Rep2_M37_raw.fastq.gz:md5,f51404578ae7128f53977bc845a67bd2",
                "Mmus_C57-6J_LVR_GC_C_Rep3_M38_raw.fastq.gz:md5,02a21f7484d1f640262ce6de9a590bdc",
                "Mmus_C57-6J_LVR_GC_C_Rep4_M39_raw.fastq.gz:md5,bfc55337a37f317cbaec8377a9286df9",
                "Mmus_C57-6J_LVR_GC_C_Rep5_M40_raw.fastq.gz:md5,cf57f68aa7725941ca7831ba6dafd451",
                "Mmus_C57-6J_LVR_GC_I_Rep1_M31_raw.fastq.gz:md5,5a99df7404567ba867d52728ed81c2d5",
                "Mmus_C57-6J_LVR_GC_I_Rep2_M32_raw.fastq.gz:md5,f0a0442543557b40d996de43f93b7146"
            ],
            [
                "Mmus_C57-6J_LVR_FLT_C_Rep1_M25_trimmed.fastq.gz:md5,d95ca667755c47327c2461201584d80f",
                "Mmus_C57-6J_LVR_FLT_C_Rep2_M26_trimmed.fastq.gz:md5,54089be1a8e415c9d6e3601dd895f407",
                "Mmus_C57-6J_LVR_FLT_C_Rep3_M27_trimmed.fastq.gz:md5,05d19501daf29aed84531b63b82f82a7",
                "Mmus_C57-6J_LVR_FLT_C_Rep4_M28_trimmed.fastq.gz:md5,b32116bf5e6385d26d068647e53c921f",
                "Mmus_C57-6J_LVR_FLT_C_Rep5_M30_trimmed.fastq.gz:md5,53dd7a9e5c98e001c34d7e0f0687ed32",
                "Mmus_C57-6J_LVR_FLT_I_Rep1_M21_trimmed.fastq.gz:md5,34ce059e22d5d4f0c25422f3616c7043",
                "Mmus_C57-6J_LVR_FLT_I_Rep2_M22_trimmed.fastq.gz:md5,da666ada4aa14919ea5fc0f28f5f4d12",
                "Mmus_C57-6J_LVR_GC_C_Rep1_M36_trimmed.fastq.gz:md5,1d7bfee51f4f5a487ad8359b37dbfc68",
                "Mmus_C57-6J_LVR_GC_C_Rep2_M37_trimmed.fastq.gz:md5,314d3a8977e6cb6095dc7c73c4fa0edd",
                "Mmus_C57-6J_LVR_GC_C_Rep3_M38_trimmed.fastq.gz:md5,2a854e1600889c9ef538251a7ff3988f",
                "Mmus_C57-6J_LVR_GC_C_Rep4_M39_trimmed.fastq.gz:md5,25d7415501016bdc4b7645008685affc",
                "Mmus_C57-6J_LVR_GC_C_Rep5_M40_trimmed.fastq.gz:md5,527e01a7290cb2b4799ff1495013b196",
                "Mmus_C57-6J_LVR_GC_I_Rep1_M31_trimmed.fastq.gz:md5,843427c45e8da0ea4358b2a6e334f635",
                "Mmus_C57-6J_LVR_GC_I_Rep2_M32_trimmed.fastq.gz:md5,fbf25f70a02ba4ec574b4a65300fefad"
            ],
            "STAR_NumNonZeroGenes.csv:md5,f6c2556ebcc175a140ca678db3eacdd6",
            "STAR_Unnormalized_Counts.csv:md5,38e39ecc85494a50679744687f3dbc1d",
            "Mmus_C57-6J_LVR_GC_C_Rep1_M36.genes.results:md5,0026606fc4edd86c12ce9f10dfcc94ef",
            "Mmus_C57-6J_LVR_GC_C_Rep1_M36.isoforms.results:md5,e0abe12f204c86024290f22b64c4d5f0",
            "RSEM_NumNonZeroGenes.csv:md5,d5a0492b5126e3da33aef5bf91d949b8",
            "RSEM_Unnormalized_Counts.csv:md5,1a877f2654556aa697b235e0d239ed08",
            [
                "Normalized_Counts.csv:md5,5d058c386aa82785f2e67c617b4aa409",
                "RSEM_Unnormalized_Counts.csv:md5,b30807d931a291daa6fdf70c8aee9694"
            ],
            [
                "SampleTable.csv:md5,db67009ac75a7fef281ced557352a2de",
                "contrasts.csv:md5,04924e148dc55056556ddcc89af862e7",
                "differential_expression.csv:md5,af9dcf3117e81c98a3707ebd798d4829",
                "visualization_PCA_table.csv:md5,e6af717c612842523e384e2a65d0a058",
                "visualization_output_table.csv:md5,fbeb584d9ee7592d07f4a069303c208e"
            ],
            [
                "software_versions.md:md5,cba543409b67025633f30daecfd65df3"
            ],
            "Mmus_C57-6J_LVR_GC_C_Rep5_M40.geneBodyCoverage.txt:md5,536cb073e5342fc4b2e7329e6bb130fe",
            "Mmus_C57-6J_LVR_GC_C_Rep5_M40_infer_expt.out:md5,cf4b9b80507493cd8232113620b4a765",
            "Mmus_C57-6J_LVR_GC_I_Rep2_M32_read_dist.out:md5,fe7bfe2aa0a2774fa2754ee0485c8b6e"
        ],
        "timestamp": "2023-05-09T20:33:40+0000"
    }
}